Valigator.Core/Core/Model.cs

```diff
@@ -244,19 +254,26 @@ private static Expression VerifyDataProperty(Expression modelExpression, Propert
 		{
 			var methods = GetVerifySupportMethods(property.PropertyType);
 
-			var dataProperty = Expression.Property(modelExpression, property);
+			var propertyParameter = Expression.Constant(property);
```

Comment:
Call the underlying getters and setters instead of relying on it being a property.